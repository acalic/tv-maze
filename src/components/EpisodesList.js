import React from "react";
import { useSelector } from "react-redux";
import { Row, Col, Image, Accordion, Badge } from "react-bootstrap";
import { removeHTML, dateTimeFormat } from "@utils/functions";

const EpisodesList = ({ episodes }) => {
  const localTimezone = useSelector((state) => state.timezone.localTimezone);

  return (
    <>
      {episodes.map((episode, i) => (
        <Accordion key={"episode-" + i} className="accordion-list mb-2">
          <Accordion.Item eventKey={i}>
            <Accordion.Header>
              Episode {episode.number} - {episode.name}
            </Accordion.Header>
            <Accordion.Body className="text-light">
              <Row>
                <Col>
                  <div className="text-muted mb-3">
                    Aired: {dateTimeFormat(episode.airstamp, localTimezone)}
                    <Badge size="sm" bg="light" className="mt-3 text-dark">
                      Local timezone: {localTimezone ? "ON" : "OFF"}
                    </Badge>
                  </div>
                  Summary:
                  {episode?.summary ? (
                    <div>{removeHTML(episode?.summary)}</div>
                  ) : (
                    <div>
                      We don't have a summary for Episode {episode.number} yet!
                    </div>
                  )}
                </Col>
                {episode?.image?.medium && (
                  <Col>
                    <Image
                      className="fluid page-show-details-img"
                      src={episode.image.medium}
                    ></Image>
                  </Col>
                )}
              </Row>
            </Accordion.Body>
          </Accordion.Item>
        </Accordion>
      ))}
    </>
  );
};

export default EpisodesList;
