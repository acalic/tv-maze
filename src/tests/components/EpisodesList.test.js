import React from "react";
import EpisodesList from "@components/EpisodesList";
import { render, screen } from "@testing-library/react";

import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

import { episodes } from "../fixtures";

describe("should test EpisodesList component", () => {
  const initialState = {
    timezone: {
      localTimezone: false,
    },
  };
  const mockStore = configureStore();
  let store;

  it("renders the EpisodesList component", () => {
    store = mockStore(initialState);

    const wrapper = render(
      <Provider store={store}>
        <EpisodesList episodes={episodes} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(screen.getByText(/Der Weg des Geldes/i)).toBeInTheDocument();
  });
});
