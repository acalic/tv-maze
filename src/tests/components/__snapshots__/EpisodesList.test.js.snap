// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test EpisodesList component renders the EpisodesList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="accordion-list mb-2 accordion"
      >
        <div
          class="accordion-item"
        >
          <h2
            class="accordion-header"
          >
            <button
              aria-expanded="false"
              class="accordion-button collapsed"
              type="button"
            >
              Episode 
              1
               - 
              Ali
            </button>
          </h2>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="text-light accordion-body"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  <div
                    class="text-muted mb-3"
                  >
                    Aired: 
                    2019/11/07 20:00
                    <span
                      class="mt-3 text-dark badge bg-light"
                    >
                      Local timezone: 
                      OFF
                    </span>
                  </div>
                  Summary:
                  <div>
                    We don't have a summary for Episode 
                    1
                     yet!
                  </div>
                </div>
                <div
                  class="col"
                >
                  <img
                    class="fluid page-show-details-img"
                    src="https://static.tvmaze.com/uploads/images/medium_landscape/223/559423.jpg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="accordion-list mb-2 accordion"
      >
        <div
          class="accordion-item"
        >
          <h2
            class="accordion-header"
          >
            <button
              aria-expanded="false"
              class="accordion-button collapsed"
              type="button"
            >
              Episode 
              2
               - 
              Der Weg des Geldes
            </button>
          </h2>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="text-light accordion-body"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  <div
                    class="text-muted mb-3"
                  >
                    Aired: 
                    2019/11/14 20:00
                    <span
                      class="mt-3 text-dark badge bg-light"
                    >
                      Local timezone: 
                      OFF
                    </span>
                  </div>
                  Summary:
                  <div>
                    We don't have a summary for Episode 
                    2
                     yet!
                  </div>
                </div>
                <div
                  class="col"
                >
                  <img
                    class="fluid page-show-details-img"
                    src="https://static.tvmaze.com/uploads/images/medium_landscape/224/562018.jpg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="accordion-list mb-2 accordion"
      >
        <div
          class="accordion-item"
        >
          <h2
            class="accordion-header"
          >
            <button
              aria-expanded="false"
              class="accordion-button collapsed"
              type="button"
            >
              Episode 
              3
               - 
              Vertrauter Feind
            </button>
          </h2>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="text-light accordion-body"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  <div
                    class="text-muted mb-3"
                  >
                    Aired: 
                    2019/11/21 20:00
                    <span
                      class="mt-3 text-dark badge bg-light"
                    >
                      Local timezone: 
                      OFF
                    </span>
                  </div>
                  Summary:
                  <div>
                    We don't have a summary for Episode 
                    3
                     yet!
                  </div>
                </div>
                <div
                  class="col"
                >
                  <img
                    class="fluid page-show-details-img"
                    src="https://static.tvmaze.com/uploads/images/medium_landscape/227/568939.jpg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="accordion-list mb-2 accordion"
    >
      <div
        class="accordion-item"
      >
        <h2
          class="accordion-header"
        >
          <button
            aria-expanded="false"
            class="accordion-button collapsed"
            type="button"
          >
            Episode 
            1
             - 
            Ali
          </button>
        </h2>
        <div
          class="accordion-collapse collapse"
        >
          <div
            class="text-light accordion-body"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <div
                  class="text-muted mb-3"
                >
                  Aired: 
                  2019/11/07 20:00
                  <span
                    class="mt-3 text-dark badge bg-light"
                  >
                    Local timezone: 
                    OFF
                  </span>
                </div>
                Summary:
                <div>
                  We don't have a summary for Episode 
                  1
                   yet!
                </div>
              </div>
              <div
                class="col"
              >
                <img
                  class="fluid page-show-details-img"
                  src="https://static.tvmaze.com/uploads/images/medium_landscape/223/559423.jpg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="accordion-list mb-2 accordion"
    >
      <div
        class="accordion-item"
      >
        <h2
          class="accordion-header"
        >
          <button
            aria-expanded="false"
            class="accordion-button collapsed"
            type="button"
          >
            Episode 
            2
             - 
            Der Weg des Geldes
          </button>
        </h2>
        <div
          class="accordion-collapse collapse"
        >
          <div
            class="text-light accordion-body"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <div
                  class="text-muted mb-3"
                >
                  Aired: 
                  2019/11/14 20:00
                  <span
                    class="mt-3 text-dark badge bg-light"
                  >
                    Local timezone: 
                    OFF
                  </span>
                </div>
                Summary:
                <div>
                  We don't have a summary for Episode 
                  2
                   yet!
                </div>
              </div>
              <div
                class="col"
              >
                <img
                  class="fluid page-show-details-img"
                  src="https://static.tvmaze.com/uploads/images/medium_landscape/224/562018.jpg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="accordion-list mb-2 accordion"
    >
      <div
        class="accordion-item"
      >
        <h2
          class="accordion-header"
        >
          <button
            aria-expanded="false"
            class="accordion-button collapsed"
            type="button"
          >
            Episode 
            3
             - 
            Vertrauter Feind
          </button>
        </h2>
        <div
          class="accordion-collapse collapse"
        >
          <div
            class="text-light accordion-body"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <div
                  class="text-muted mb-3"
                >
                  Aired: 
                  2019/11/21 20:00
                  <span
                    class="mt-3 text-dark badge bg-light"
                  >
                    Local timezone: 
                    OFF
                  </span>
                </div>
                Summary:
                <div>
                  We don't have a summary for Episode 
                  3
                   yet!
                </div>
              </div>
              <div
                class="col"
              >
                <img
                  class="fluid page-show-details-img"
                  src="https://static.tvmaze.com/uploads/images/medium_landscape/227/568939.jpg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
